components/director/internal/tenantmapping/cert_service_context_provider.go

```diff
@@ -49,39 +50,73 @@ func (m *certServiceContextProvider) GetObjectContext(ctx context.Context, reqDa
 	// This if is needed to separate the director from ord flow because for the director flow we need to use the internal ID of the subaccount
 	// whereas in the ord flow we expect external IDs in order ord views to work properly(using Automatic Scenario Assignments)
 	log.C(ctx).Infof("Matched component name is %s", matchedComponentName[0])
-	if matchedComponentName[0] == "director" { // Director Flow, do the tenant conversion
-		scopes := "runtime:read runtime:write tenant:read"
 
-		log.C(ctx).Infof("Getting the tenant with external ID: %s", externalTenantID)
-		tenantMapping, err := m.tenantRepo.GetByExternalTenant(ctx, externalTenantID)
-		if err != nil {
-			if apperrors.IsNotFoundError(err) {
-				log.C(ctx).Warningf("Could not find tenant with external ID: %s, error: %s", externalTenantID, err.Error())
+	// the authID in this flow is an OU selected by the Connector
+	externalTenantID := authDetails.AuthID
+
+	if matchedComponentName[0] != "director" { // ORD Flow, set the external tenant ID both for internal and external tenants
+		objCtx := NewObjectContext(NewTenantContext(externalTenantID, externalTenantID), p.tenantKeys, "", authDetails.Region, "", authDetails.AuthID, authDetails.AuthFlow, consumer.Runtime, CertServiceObjectContextProvider)
+		log.C(ctx).Infof("Successfully got object context: %+v", objCtx)
+		return objCtx, nil
+	}
```

Comment:
The logic of the `GetObjectContext` up to this line seems like it is related to ORD only.
My proposal is to try and extract it either in another ObjectContextProvider with matcher for the component or at least move it in a separate method in this context provider.

Reply:
It's removed in the "Storage Redesign" PR, we'll remove it here as well